operators/pkg/controller/kibana/deployment_control.go

```diff
@@ -64,16 +59,18 @@ func ReconcileDeployment(c k8s.Client, s *runtime.Scheme, expected appsv1.Deploy
 		Expected:   &expected,
 		Reconciled: reconciled,
 		NeedsUpdate: func() bool {
-			return !reflect.DeepEqual(expected.Spec.Selector, reconciled.Spec.Selector) ||
+			expectedContainer := pod.GetKibanaContainer(expected.Spec.Template.Spec)
+			actualContainer := pod.GetKibanaContainer(reconciled.Spec.Template.Spec)
+			return actualContainer == nil ||
+				!reflect.DeepEqual(expected.Spec.Selector, reconciled.Spec.Selector) ||
 				!reflect.DeepEqual(expected.Spec.Replicas, reconciled.Spec.Replicas) ||
 				!reflect.DeepEqual(expected.Spec.Template.Spec.Affinity, reconciled.Spec.Template.Spec.Affinity) ||
 				!reflect.DeepEqual(expected.Spec.Template.ObjectMeta, reconciled.Spec.Template.ObjectMeta) ||
-				!reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Name, reconciled.Spec.Template.Spec.Containers[0].Name) ||
-				!reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Env, reconciled.Spec.Template.Spec.Containers[0].Env) ||
-				!reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Image, reconciled.Spec.Template.Spec.Containers[0].Image) ||
-				!reflect.DeepEqual(expected.Spec.Template.Spec.Containers[0].Resources, reconciled.Spec.Template.Spec.Containers[0].Resources)
+				!reflect.DeepEqual(expectedContainer.Name, actualContainer.Name) ||
+				!reflect.DeepEqual(expectedContainer.Env, actualContainer.Env) ||
+				!reflect.DeepEqual(expectedContainer.Image, actualContainer.Image) ||
+				!reflect.DeepEqual(expectedContainer.Resources, actualContainer.Resources)
 			// TODO: do something better than reflect.DeepEqual above?
```

Comment:
This is a bit out-of-scope of this PR. I think we miss comparing several fields that the user can fill. We definitively need to improve this `NeedsUpdate` function.

Reply:
Yes ! I started doing it in this PR then realized it should be better off to another PR fixing issue https://github.com/elastic/cloud-on-k8s/issues/772